src/main/webapp/app/tasks/task-list/task-list.component.html

```diff
@@ -1,9 +1,48 @@
 <mat-list data-cy="task-list">
   <mat-list-item *ngFor="let task of tasks" class="mat-elevation-z1">
     <mat-icon mat-list-icon>assignment</mat-icon>
-    <h4 mat-line>{{task.name}}</h4>
+    <h4 mat-line>{{ task.name }}</h4>
     <button mat-icon-button aria-label="Delete task" color="primary">
-      <mat-icon aria-label="Delete task" (click)="delete(task)">delete</mat-icon>
+      <mat-icon aria-label="Delete task" (click)="delete(task)"
+        >delete</mat-icon
+      >
+    </button>
+    <label>
+      <input
+        aria-label="mark task as done"
+        type="checkbox"
+        [color]="'primary'"
+        [checked]="task.done"
+        (change)="setIsDone($event, task)"
+      />
+    </label>
+    <button mat-icon-button aria-label="Done task">
+      <mat-icon
+        aria-label="mark task as done"
+        (change)="setIsDone($event)"
+        [ngStyle]="{ color: task.status === 'done' ? 'green' : 'primary' }"
+        >done</mat-icon
+      >
     </button>
   </mat-list-item>
+  <button mat-button aria-label="Delete task" (click)="deleteDoneTasks(task)">
```

Comment:
this doesn't need the task to be passed and indeed the task is not part of the signature/a paramater

Reply:
So this should be filled with the "done.task" or maybe just be empty?